### PR TITLE
 Fix bugs of UserStore delay

### PR DESCRIPTION
**Pull request by zycbobby**

### CODE REVIEW

File: app/components/InstantMessage/ImMain/Main.jsx [hunk 1]
```diff
@@ -84,23 +103,15 @@ module.exports = React.createClass({
...
-        if (!SocketStore.getSocket()) {
-          _channels = _init();
-          InitAction.init(_channels, LoginStore.getUser());
-        }
-        // validate channelIdToGo
-        if (isChannelIdValid(channelIdToGo)) {
-          ChannelAction.changeChannel(channelIdToGo, LoginStore.getUser());
-        }
+      if (UserStore.getUsersArray() && UserStore.getUsersArray().length === 0) {
+        // wait for the users array arrive
```
Comment (LeonLiuY) [inline app/components/InstantMessage/ImMain/Main.jsx:107]: Should it be? 

```
if (!UserStore.getUsersArray() || UserStore.getUsersArray().length === 0) {
```

Comment (LeonLiuY) [inline app/components/InstantMessage/ImMain/Main.jsx:107]: To make it more reliable & expressive, you can add a variable `initialized` in UserStore, then use `UserStore.isInitialized()`

Comment (zycbobby) [inline app/components/InstantMessage/ImMain/Main.jsx:107]: I think so

**Comment by zycbobby** [comment]
@liuyang1024  Sorry, L149 in Main.jsx will have problem in the channel title, please dont deploy it now

**Pull request merged**